info.limpet.stackedcharts.ui/src/info/limpet/stackedcharts/ui/editor/drop/DatasetToAxisDropTargetListener.java

```diff
@@ -38,41 +41,47 @@ public DatasetToAxisDropTargetListener(GraphicalViewer viewer)
   @Override
   public void dropAccept(DropTargetEvent event)
   {
-    if (event.detail == DND.DROP_DEFAULT)
-    {
-      if ((event.operations & DND.DROP_COPY) != 0)
-      {
-        event.detail = DND.DROP_COPY;
-      }
-      else
-      {
-        event.detail = DND.DROP_COPY;
-      }
-    }
-    for (int i = 0; i < event.dataTypes.length; i++)
+  }
+
+  protected boolean canDrop(final AxisEditPart axis, ISelection selection)
+  {
+    boolean canDrop = false;
+    AdapterRegistry adapter = new AdapterRegistry();
+    if (selection instanceof StructuredSelection)
     {
-      if (LocalSelectionTransfer.getTransfer().isSupportedType(
-          event.dataTypes[i]))
+      for (Object obj : ((StructuredSelection) selection).toArray())
       {
-        event.currentDataType = event.dataTypes[i];
-
-        if (event.detail != DND.DROP_COPY)
+        if (adapter.canConvert(obj))
         {
-          event.detail = DND.DROP_COPY;
+          final List<Dataset> validDatasets = new ArrayList<Dataset>();
+          List<Dataset> convert = adapter.convert(obj);
+          if (convert.size() == 0)
+          {
+            continue;
+          }
+          for (Dataset dataset : convert)
+          {
+
+            addIfPossible(axis, validDatasets, dataset);
+          }
```

Comment:
Hmm, this is the **canDrop()** method.  "addIfPossible()" actually adds the dataset to the axis.  I'm sure canDrop isn't the place to actually add the dataset. I'm sure **drop()** is.